mentat/diff_context.py

```diff
@@ -107,7 +107,8 @@ def files(self) -> list[Path]:
         if self._files_cache is None:
             if self.target == "HEAD" and not check_head_exists():
                 return []  # A new repo without any commits
-            self._files_cache = get_files_in_diff(self.target)
+            diff_files = get_files_in_diff(self.target)
+            self._files_cache = [f.resolve() for f in diff_files]
```

Comment:
We need to use git_root / f, not f.resolve() (since these paths are relative to the git root, not the cwd). I also think it would be cleaner to do it in one line (instead of making this diff_files varialbe), but that's up to you